metpy/calc/tests/test_tools.py

```diff
@@ -28,14 +29,46 @@ def test_nearest_intersection_idx():
     assert_array_equal(truth, nearest_intersection_idx(y1, y2))
 
 
-def test_find_intersections():
+intersection_directions = ['all', 'increasing', 'decreasing', 'decdreasing']
+
+
+@pytest.mark.parametrize('direction', intersection_directions)
```

Comment:
How about adding either the truth data itself, or at least a slice/index selecting it from the master array as an additional parameter? Then you can eliminate the nasty if chain in the test. Just make the invalid direction test its own.